iceoryx_hoofs/include/iceoryx_hoofs/cxx/variant.hpp

```diff
@@ -273,12 +277,33 @@ class variant
     static void error_message(const char* source, const char* msg) noexcept;
 
     void call_element_destructor() noexcept;
+
+    template <typename... Ts>
+    friend constexpr bool operator==(const variant<Ts...>& lhs, const variant<Ts...>& rhs);
+    template <typename... Ts>
+    friend constexpr bool operator!=(const variant<Ts...>& lhs, const variant<Ts...>& rhs);
 };
 
 /// @brief returns true if the variant holds a given type T, otherwise false
 template <typename T, typename... Types>
 constexpr bool holds_alternative(const variant<Types...>& variant) noexcept;
 
+/// @brief equality check for two distinct variant types
+/// @tparam Types variadic number of types which can be stored in the variant
+/// @param[in] lhs left side of the comparision
```

Comment:
```suggestion
/// @param[in] lhs left side of the comparison
```
Same typo in several lines below.